### PR TITLE
Add multigroups tests cases for `test_assign_groups_guess`

### PR DESCRIPTION
**Pull request by juliamagan**
|Related issue|
|-------------|
|     #3913        |

### Added

- Added `test_guess_multigroups`

### Updated

- Updated `remove_cluster_agents`: accept agent ids
- Updated provisioning: deploy two agents

---

## Testing performed

| Tester             | Test path | Jenkins | Local   | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |    `system/test_cluster/test_agent_groups/test_assign_groups_guess.py`       | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10881746/R1-test-assign-groups-guess.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10881749/R2-test-assign-groups-guess.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10882268/R3-test-assign-groups-guess.zip) |        | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |           | Nothing to highlight |

### Affected tests

| Tester             | Test path | Jenkins | Local   | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |    `system/test_cluster/test_agent_groups/test_assign_agent_to_a_group.py`       | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10881712/R1_test_assign_agent_to_a_group.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10881713/R2_test_assign_agent_to_a_group.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10881715/R3_test_assign_agent_to_a_group.zip) |        | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |           | Nothing to highlight |

| Tester             | Test path | Jenkins | Local   | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |    `system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_by_tool.py`       | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10881719/R1_test_assign_agent_to_a_group_by_tool.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10881720/R2_test_assign_agent_to_a_group_by_tool.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10881723/R3_test_assign_agent_to_a_group_by_tool.zip) |        | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |           | Nothing to highlight |

### CODE REVIEW

File: tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py [hunk 1]
```diff
@@ -149,18 +150,115 @@ def test_assign_agent_to_a_group(agent_target, status_guess_agent_group, clean_e
...
+                                       after='upgrades.', before='authd.debug=2', replace=replace)
+    # Restart managers
+    restart_cluster(test_infra_managers, host_manager)
+    time.sleep(timeout)
+
+    # Register agent with agent-auth
+    agent1_ip, agent1_id, agent1_name, ag1_manager_ip = register_agent(test_infra_agents[0], target_node,
+                                                                       host_manager)
+    if n_agents == 2:
+        agent2_ip, agent2_id, agent2_name, ag2_manager_ip = register_agent(test_infra_agents[1], target_node,
```
Comment (jmv74211) [inline tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py:213]: This is not scalable

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py (hunk 1 above)
Comment (juliamagan) [reply tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py]: Done in 67da75d

File: tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py [hunk 2]
```diff
@@ -149,18 +150,115 @@ def test_assign_agent_to_a_group(agent_target, status_guess_agent_group, clean_e
...
+        - Verify that after registering it has the 'Null' group assigned.
+        - Verify that after assign group with agent-groups the change is sync with the cluster.
+    expected_output:
+        - The agent 'Agent_name' with ID 'Agent_id' belongs to groups: group_test."
+    '''
+    # Modify local_internal_options
+    replace = '\n' + remoted_guess_agent_groups + f"{status_guess_agent_group}\n"
+
+    for host in test_infra_managers:
+        host_manager.add_block_to_file(host, path=f"{WAZUH_PATH}/etc/local_internal_options.conf",
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py]: Done in 67da75d

File: tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py [hunk 3]
```diff
@@ -71,21 +71,36 @@
...
-remoted_guess_agent_groups = 'remoted.guess_agent_group='
+group_id = 'group_test'
+multigroups_id = 'default,group_test'
 # this timeout is temporality, this test will be update
 timeout = 20
 
 
+# Fixtures
+@pytest.fixture()
+def modify_internal_options(status_guess_agent_group):
```
Comment (jmv74211) [inline tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py:82]:
```suggestion
def modify_local_internal_options(status_guess_agent_group):
```

File: tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py [hunk 4]
```diff
@@ -102,19 +120,12 @@ def test_assign_agent_to_a_group(agent_target, status_guess_agent_group, clean_e
...
         - The agent 'Agent_name' with ID 'Agent_id' belongs to groups: group_test."
     '''
-    group_id = 'group_test'
-    # Modify local_internal_options
-    replace = '\n' + remoted_guess_agent_groups + f'{status_guess_agent_group}\n'
-
-    for host in test_infra_managers:
-        host_manager.add_block_to_file(host, path=f"{WAZUH_PATH}/etc/local_internal_options.conf",
-                                       after="upgrades.", before="authd.debug=2", replace=replace)
     # Restart managers
```
Comment (jmv74211) [inline tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py:123]: Update this comment

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/tools/system.py [hunk 5]
```diff
@@ -322,6 +322,22 @@ def get_stats(self, host: str, path: str):
         """
         return self.get_host(host).ansible("stat", f"path={path}")
 
+    def configure_internal_options(self, local_internal_options: dict):
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply deps/wazuh_testing/wazuh_testing/tools/system.py]: Done in bdad304

File: tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py (hunk 3 above)
Comment (juliamagan) [reply tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py]: Done in bdad304

File: tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py (hunk 4 above)
Comment (juliamagan) [reply tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py]: Done in ccaa8e3

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**